Link against an actual target name, not the generated library name.

If we link against the latter, there is no dependency between lastfm-fpclient
and fplib, which can break parallel builds starting with CMake 3.5.0.
--- CMakeLists.txt.orig	2013-10-23 16:45:46 UTC
+++ CMakeLists.txt
@@ -139,7 +139,7 @@ ADD_EXECUTABLE(lastfm-fpclient
 
 # Link the executable
 TARGET_LINK_LIBRARIES(lastfm-fpclient
-   fplib
+   fplib_shared
    fftw3f
    mad
    tag
